core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -95,13 +116,6 @@ void generateConfigClass(
             return;
         }
 
-        Map<String, String> defaults = new HashMap<>();
-        for (RunTimeConfigurationDefaultBuildItem item : runTimeDefaults) {
-            if (defaults.putIfAbsent(item.getKey(), item.getValue()) != null) {
-                throw new IllegalStateException("More than one default value for " + item.getKey() + " was produced");
```

Comment:
I have stumbled over this validation a few times in the past, since I tried to use this mechanism to set up default values for various config properties which then sometimes conflicted with properties set by other extensions. Specifically, I tried to set a default for the dev-services Docker image name this way.

It seems like this validation no longer exists now and that it may be a case of last build item wins. Meanwhile I found a different solution using a custom config source, but due to the indeterminism it may make sense to keep this validation in place somewhere. Thoughts?